docs/user-guide.rst

```diff
@@ -544,6 +544,246 @@ For AWS providers, use :meth:`aws.Credentials.from_info
+subject tokens and exchange them for GCP access tokens. The configuration file
+can be generated by using the `gcloud CLI`_.
+
+The Auth library can retrieve external subject tokens from a local file
+location (file-sourced credentials), from a local server (URL-sourced
+credentials) or by calling an executable (executable-sourced credentials).
+
+File-sourced credentials For file-sourced credentials, a background process
+needs to be continuously refreshing the file location with a new subject token
+prior to expiration. For tokens with one hour lifetimes, the token needs to be
+updated in the file every hour. The token can be stored directly as plain text
+or in JSON format.
+
+To generate a file-sourced OIDC configuration, run the following command:
+
+.. code-block:: bash
+
+    # Generate an OIDC configuration file for file-sourced credentials.
+    gcloud iam workforce-pools create-cred-config \
+        locations/global/workforcePools/$WORKFORCE_POOL_ID/providers/$PROVIDER_ID \
+        --subject-token-type=urn:ietf:params:oauth:token-type:id_token \
+        --credential-source-file=$PATH_TO_OIDC_ID_TOKEN \
+        --workforce-pool-user-project=$WORKFORCE_POOL_USER_PROJECT \
+        # Optional arguments for file types. Default is "text":
+        # --credential-source-type "json" \
+        # Optional argument for the field that contains the OIDC credential.
+        # This is required for json.
+        # --credential-source-field-name "id_token" \
+        --output-file=/path/to/generated/config.json
+
+Where the following variables need to be substituted:
+
+* ``$WORKFORCE_POOL_ID``: The workforce pool ID.
+* ``$PROVIDER_ID``: The provider ID.
+* ``$PATH_TO_OIDC_ID_TOKEN``: The file path used to retrieve the OIDC token.
+* ``$WORKFORCE_POOL_USER_PROJECT``: The project number associated with the
+  `workforce pools user project`_.
+
+To generate a file-sourced SAML configuration, run the following command:
+
+.. code-block:: bash
+
+    # Generate a SAML configuration file for file-sourced credentials.
+    gcloud iam workforce-pools create-cred-config \
+        locations/global/workforcePools/$WORKFORCE_POOL_ID/providers/$PROVIDER_ID \
+        --credential-source-file=$PATH_TO_SAML_ASSERTION \
+        --subject-token-type=urn:ietf:params:oauth:token-type:saml2 \
+        --workforce-pool-user-project=$WORKFORCE_POOL_USER_PROJECT \
+        --output-file=/path/to/generated/config.json
+
+Where the following variables need to be substituted:
+
+* ``$WORKFORCE_POOL_ID``: The workforce pool ID.
+* ``$PROVIDER_ID``: The provider ID.
+* ``$PATH_TO_SAML_ASSERTION``: The file path used to retrieve the
+  base64-encoded SAML assertion.
+* ``$WORKFORCE_POOL_USER_PROJECT``: The project number associated with the
+  `workforce pools user project`_.
+
+These commands generate the configuration file in the specified output file.
+
+URL-sourced credentials For URL-sourced credentials, a local server needs to
```

Comment:
did you intend this to be **URL-sourced credentials** For URL-sourced credentials?

Same for file sourced

Reply:
Took a look at the source doc in Java library. That was a title which didn't outlined. I'll fix here